File: 251004/Symonik/issue-service/src/main/java/by/symonik/issue_service/entity/service/LabelService.java
package by.symonik.issue_service.entity.service;

import by.symonik.issue_service.dto.label.LabelRequestTo;
import by.symonik.issue_service.dto.label.LabelResponseTo;
import by.symonik.issue_service.dto.label.LabelUpdateRequestTo;

import java.util.List;

public interface LabelService {
    LabelResponseTo create(LabelRequestTo labelRequestTo);

    List<LabelResponseTo> readAll();

    LabelResponseTo readById(Long id);

    LabelResponseTo update(LabelUpdateRequestTo labelUpdateRequestTo);

    void deleteById(Long id);
}

File: 251003/Golovin/appLab2/src/main/kotlin/com/example/eger/dc/lab2/service/LabelService.kt
package com.example.eger.dc.lab2.service

import com.example.eger.dc.lab2.dto.request.LabelRequestTo
import com.example.eger.dc.lab2.dto.request.LabelRequestToId
import com.example.eger.dc.lab2.dto.response.LabelResponseTo

interface LabelService {
	suspend fun create(requestTo: LabelRequestTo?): LabelResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getAll(): List<LabelResponseTo>

	suspend fun getById(id: Long): LabelResponseTo?

	suspend fun update(requestTo: LabelRequestToId?): LabelResponseTo?
}